Animal Competition/Java/Environment.java

```diff
@@ -0,0 +1,23 @@
+public abstract class Environment {
+
+  public abstract boolean hasTrees();
+
+  public abstract boolean hasWater();
+
+  public abstract boolean hasGrass();
+
+  public abstract boolean hasSand();
+
+  public abstract boolean hasPlants();
+
+  public abstract boolean hasOtherAnimals();
+
+  public abstract boolean hasFood();
+
+  public abstract boolean isHot();
```

Comment:
Everything seems so black-and-white in this game... have you consider using continuous values (numbers) rather than just booleans for these properties... like `Temperature`?